packages/core/src/machines/actors/auth/resetPassword.ts

```diff
@@ -0,0 +1,115 @@
+import { createMachine, assign, sendUpdate } from 'xstate';
+
+import { AuthEvent, ResetPasswordContext } from '../../../types';
+import { Auth } from 'aws-amplify';
+
+export const resetPasswordActor = createMachine<
+  ResetPasswordContext,
+  AuthEvent
+>(
+  {
+    id: 'resetPasswordActor',
+    initial: 'resetPassword',
+    states: {
+      resetPassword: {
+        initial: 'edit',
+        exit: ['clearFormValues', 'clearError'],
+        states: {
+          edit: {
+            entry: sendUpdate(),
+            on: {
+              SUBMIT: 'submit',
+              CHANGE: { actions: 'handleInput' },
+            },
+          },
+          submit: {
+            entry: [sendUpdate(), 'setUsername', 'clearError'],
+            invoke: {
+              src: 'resetPassword',
+              onDone: {
+                target: '#resetPasswordActor.confirmResetPassword',
+              },
+              onError: {
+                actions: ['setRemoteError'],
+                target: 'edit',
+              },
+            },
+          },
+        },
+      },
+      confirmResetPassword: {
+        initial: 'edit',
+        exit: ['clearFormValues', 'clearError', 'clearUsername'],
+        states: {
+          edit: {
+            entry: sendUpdate(),
+            on: {
+              SUBMIT: 'submit',
+              RESEND: 'resendCode',
+              CHANGE: { actions: 'handleInput' },
+            },
+          },
+          resendCode: {
+            entry: ['clearError', sendUpdate()],
+            invoke: {
+              src: 'resetPassword',
+              onDone: { target: 'edit' },
+              onError: {
+                actions: 'setRemoteError',
+                target: 'edit',
+              },
+            },
+          },
+          submit: {
+            entry: ['clearError', sendUpdate()],
+            invoke: {
+              src: 'confirmResetPassword',
+              onDone: {
+                actions: 'clearUsername',
+                target: '#resetPasswordActor.resolved',
+              },
+              onError: {
+                actions: 'setRemoteError',
+                target: 'edit',
+              },
+            },
+          },
+        },
+      },
+      resolved: { type: 'final' },
+    },
+  },
+  {
+    actions: {
+      setRemoteError: assign({
+        remoteError: (_, event) => event.data?.message || event.data,
+      }),
+      setUsername: assign({
+        username: (context) => context.formValues.username,
+      }),
+      handleInput: assign({
```

Comment:
What do you think about changing this action name to `handleChange`?

Reply:
Ah, that's a good suggestion. I personally like `handleInput` more because it indicates that it's the "input" that's being handled. I think `handleChange` isn't that clear about which change it's handling.  

Maybe `updateInput`, or `setInput` can work better?

Reply:
Now that I think about it, just `INPUT` events makes the most sense 😄  Let me try that on a future pr, but also open to discussions.